### ISSUE TITLE
[Bug]: There is an error with max_length

### ISSUE THREAD

**Issue by kongbo96** [milvus-io/milvus#23050]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
pymilvus 2.2.1
milvus2.1.4
```

### Current Behavior

![image](https://user-images.githubusercontent.com/42367656/222624671-f055d6be-8f30-4ad3-a21f-897ef1e0d9ef.png)

![image](https://user-images.githubusercontent.com/42367656/222624731-572a41ae-574b-464f-9669-1975756425f3.png)
The length of “query“” column data cannot be greater than 10, but the length of “t_deadline_ts“ column data can be greater than 10.

It's ridiculous.

**Comment by yanliang567** [comment]
sounds like an attu issue.
/assign @shanghaikid 
/unassign

**Comment by shanghaikid** [comment]
Yeah, attu will never modify the input/output any data. @yanliang567

**Comment by yanliang567** [comment]
@kongbo96 could you please share some lines of code to reproduce the issue? i fail to reproduce it in house.

/assign @kongbo96

**Comment by kongbo96** [comment]
> @kongbo96 could you please share some lines of code to reproduce the issue? i fail to reproduce it in house.


```
#!/usr/bin/env python
import argparse
import pandas as pd
from pymilvus import (
    connections,
    utility,
    FieldSchema, CollectionSchema, DataType,
    Collection,
)
from timeit import default_timer as timer


def main():
    parser = argparse.ArgumentParser()

    parser.add_argument("--host", required=True, type=str)
    parser.add_argument("--port", required=True, type=str)
    parser.add_argument("--collection", required=True, type=str)
    parser.add_argument("--rebuild_collection", required=True, type=int)
    parser.add_argument("--new_pk", required=True, type=str)
    parser.add_argument("--in_parquet", required=True, type=str)
    parser.add_argument("--cols", required=True, type=str)
    parser.add_argument("--max_lens", type=str)
    parser.add_argument("--vec_col", required=True, type=str)

    args = parser.parse_args()

    HOST = args.host
    PORT = args.port
    COLLECTION = args.collection
    REBUILD_COLLECTION = args.rebuild_collection
    IN_PARQUET = args.in_parquet
    NEW_PK = args.new_pk
    MAX_LENS = [str(l) for l in args.max_lens.split(',')] if args.max_lens is not None else 'AUTO'
    COLS = args.cols.split(',')
    VEC_COL = args.vec_col

    print('HOST:', HOST)
    print('PORT:', PORT)
    print('COLLECTION:', COLLECTION)
    print('REBUILD_COLLECTION:', REBUILD_COLLECTION)
    print('IN_PARQUET:', IN_PARQUET)
    print('NEW_PK:', NEW_PK)
    print('COLS:', COLS)
    print('MAX_LENS:', MAX_LENS)
    print('VEC_COL:', VEC_COL)

    df = pd.read_parquet(IN_PARQUET)
    max_lens = [df[col].str.len().max() for col in COLS] if MAX_LENS == 'AUTO' else MAX_LENS
    if MAX_LENS == 'AUTO':
        print('COL_MAX_LENS:', ' '.join([str(max_len) for max_len in max_lens]))
    vec_len = len(df.iloc[0][VEC_COL])
    print('VEC_LENGTH:', vec_len)

    # 建立连接
    connections.connect(alias='default', host=HOST, port=PORT)
    if REBUILD_COLLECTION == 1:
        if utility.has_collection(COLLECTION):
            utility.drop_collection(COLLECTION)

    # 建立schema
    fields = [FieldSchema(name=NEW_PK, dtype=DataType.INT64, is_primary=True, auto_id=True)]
    fields.extend([FieldSchema(name=col, dtype=DataType.VARCHAR, max_length=max_len) for col, max_len in zip(COLS, max_lens)])
    fields.append(FieldSchema(name=VEC_COL, dtype=DataType.FLOAT_VECTOR, dim=vec_len))
    print('FIELDS:', fields)
    schema = CollectionSchema(fields)
    collection = Collection(name=COLLECTION, schema=schema)

    # 准备要传输的数据
    ALL_COLS = COLS + [VEC_COL]
    print(ALL_COLS)
    df = df[ALL_COLS]
    df_dict = df.to_dict(orient='list')
    entities = [df_dict[col] for col in ALL_COLS]

    # 传输数据
    begin = timer()
    result = collection.insert(entities)
    end = timer()
    print('Element Written to Milvus:', result.insert_count)
    print('Execution: {:.2f} seconds'.format(end - begin))
    collection.load()


if __name__ == '__main__':
    main()

```
This is the complete code.
@yanliang567

**Comment by yanliang567** [comment]
/assign @NicoYuan1986 
could you please help to reproduce it in house?

**Comment by NicoYuan1986** [comment]
Reproduced.
pymilvus:2.2.1
milvus:2.1.4
code:
```python
fields = [
    FieldSchema(name='id', dtype=DataType.INT64, is_primary=True, auto_id=True),
    FieldSchema(name='t_id', dtype=DataType.VARCHAR, max_length=32),
    FieldSchema(name='t_deadline_ts', dtype=DataType.VARCHAR, max_length=10),
    FieldSchema(name='query', dtype=DataType.VARCHAR, max_length=53),
    FieldSchema(name='query_embedding', dtype=DataType.FLOAT_VECTOR, dim=768)
]
schema = CollectionSchema(fields)
collection = Collection(name="test_length", schema=schema)
vectors = [[random.random() for _ in range(768)] for _ in range(1)]
entity = [
    ['06dndisajfhieuw73nkjsdvkjd'],
    ['111111111111111111111111'],
    ['1111111'],
    vectors
]
collection.insert(entity)
```
result: 
![image](https://user-images.githubusercontent.com/109071306/231140127-f7b17cc4-b7d6-457a-81e0-90f4859ca94e.png)
Not expected. A MilvusException is needed.

**Comment by NicoYuan1986** [comment]
Fixed with 2.2.5.
milvus: 2.2.5
pymilvus: 2.2.5
When executing the same code:
```
raise ParamError(message=f"invalid input, length of string exceeds max length. length: {len(s)}, "
pymilvus.exceptions.ParamError: <ParamError: (code=1, message=invalid input, length of string exceeds max length. length: 24, max length: 10)>
```
